client/client.go

```diff
@@ -239,6 +241,35 @@ func (c *Client) NewUnAuthRequest(ctx context.Context, method, urlStr string, bo
 	return req, nil
 }
 
+// NewUnAuthFormEncodedRequest returns content-type: application/x-www-form-urlencoded based unauth request
+func (c *Client) NewUnAuthFormEncodedRequest(ctx context.Context, method, urlStr string, body map[string]string) (*http.Request, error) {
```

Comment:
Can we try implementing same functionality in the same older func by using a flag value ?